src/sentry/notifications/notification_action/metric_alert_registry/handlers/slack_metric_alert_handler.py

```diff
@@ -45,15 +47,16 @@ def send_alert(
         if not open_period:
             raise ValueError("Open period not found")
 
-        alert_rule_serialized_response = get_alert_rule_serializer(detector)
-        incident_serialized_response = get_detailed_incident_serializer(open_period)
-
-        logger.info(
-            "notification_action.execute_via_metric_alert_handler.slack",
-            extra={
-                "action_id": alert_context.action_identifier_id,
-            },
-        )
+        alert_rule_serialized_response = None
+        incident_serialized_response = None
+        try:
+            alert_rule_serialized_response = get_alert_rule_serializer(detector)
+            incident_serialized_response = get_detailed_incident_serializer(open_period)
+        except SoftTimeLimitExceeded:
+            logger.exception(
+                "notification_action.execute_via_metric_alert_handler.slack.soft_time_limit_exceeded",
+                extra={"action_id": alert_context.action_identifier_id},
```

Comment:
Conceptually this makes sense, but I wonder if we need to be more proactive about enforcing time limits from the query side of things. Like time-bounding the shared serializer code in a consistent way that degrades the alert rule data we get when this happens. 

My concern is that this just patches over the issue, and we still have to contend with the case where we exceed the hard deadline and this task is forcibly killed before calling into notification code.